### PR TITLE
Add a bunch more tests for annotation-xml

### PR DESCRIPTION
**Pull request by gsnedders**

### CODE REVIEW

File: tree-construction/tests20.dat [hunk 1]
```diff
@@ -514,3 +514,57 @@
 |       <math annotation-xml>
 |         encoding=" text/html "
 |     <div>
+
+#data
+<math><annotation-xml> </annotation-xml>
+#errors
```
Comment (zcorpan) [inline tree-construction/tests20.dat]: Should expect some errors for these new tests, right?

File: tree-construction/tests20.dat [hunk 2]
```diff
@@ -514,3 +514,57 @@
...
+#document
+| <html>
+|   <head>
+|   <body>
+|     <math math>
+|       <math annotation-xml>
+|         <!-- foo -->
+
+#data
+<math><annotation-xml></svg>
```
Comment (zcorpan) [inline tree-construction/tests20.dat]: Maybe add an "x" at the end to see that it is inserted in the right place

File: tree-construction/tests20.dat (hunk 1 above)
Comment (gsnedders) [inline tree-construction/tests20.dat]: Yeah, we should. But AFAIK the errors in general are frequently wrong and I felt like giving up when I did this.

Comment (nolanw) [inline tree-construction/tests20.dat]: I found the errors that existed were very helpful when writing a parser, but I understand it's a pain to add and maintain them.

Comment (gsnedders) [inline tree-construction/tests20.dat]: @nolanw I just don't have anything off-hand that's up-to-date to generate them from, which is what makes it a real pain. (HTMLReader doesn't have any easy-to-run test binary, does it? That, say, takes input from either a file or stdin and dumps the test expectation format?)

Comment (nolanw) [inline tree-construction/tests20.dat]: I kinda gave up on the errors once things mostly worked ok (surprise!). That's a good idea for a lil binary though, I'll tuck that under my hat and maybe one day the errors will rain down on us all.

**Comment by zcorpan** [comment]
OK, LGTM with the "x" change

**Pull request merged**